mypy/nodes.py

```diff
@@ -3851,6 +3867,43 @@ def deserialize(cls, data: JsonDict) -> SymbolTable:
         return st
 
 
+class DataclassTransformSpec:
```

Comment:
I had two reasons for splitting this into its own (dataclass transform-specific) class:

1. We can use `node.dataclass_transform_spec is not None` to determine whether a node should act as a dataclass trasnform (ie, we can use its presence or absence as a flag).
2. We'll also need to attach these fields to `ClassDef` (or `TypeInfo`) since PEP 681 supports those as well.

Do those seem like reasonable motivation to keep it as-is for this PR?

Reply:
Yeah, it makes sense to keep it as-is.